monix/shared/src/main/scala/org/atnos/eff/addon/monix/TaskEffect.scala

```diff
@@ -0,0 +1,240 @@
+      sexs.schedule(onTimeout, t.length, t.unit)
+      Task.unsafeStartAsync(task(sexs), context, new Callback[A] {
+        override def onSuccess(value: A): Unit = {
+          val _ = promise.trySuccess(value)
+        }
+        override def onError(ex: Throwable): Unit = {
+          val _ = promise.tryFailure(ex)
+        }
+      })
+      promise.future.onComplete(callback)(context.scheduler)
+    }
+  }
+}
+
+object TimedTask {
+  final def TimedTaskApplicative: Applicative[TimedTask] = new Applicative[TimedTask] {
+    override def pure[A](x: A) = TimedTask(_ => Task.now(x))
+    override def ap[A, B](ff: TimedTask[(A) => B])(fa: TimedTask[A]) =
+      TimedTask(sexs => Task.mapBoth(ff.runNow(sexs), fa.runNow(sexs))(_ (_)))
+  }
+
+  implicit final def TimedTaskMonad: Monad[TimedTask] = new Monad[TimedTask] {
+    override def pure[A](x: A) = TimedTask(_ => Task.now(x))
+    override def flatMap[A, B](fa: TimedTask[A])(f: (A) => TimedTask[B]) =
+      TimedTask(sexs => fa.runNow(sexs).flatMap(f(_).runNow(sexs)))
+    override def tailRecM[A, B](a: A)(f: (A) => TimedTask[Either[A, B]]): TimedTask[B] =
+      TimedTask[B]({ sexs =>
+        def loop(na: A): Task[B] = f(na).runNow(sexs).flatMap(_.fold(loop, Task.now))
+        loop(a)
+      })
+  }
+
+  final def now[A](value: A): TimedTask[A] =
+    TimedTask(_ => Task.now(value))
+  implicit final def fromTask[A](task: Task[A]): TimedTask[A] =
+    TimedTask(_ => task)
+  final def fromTask[A](task: Task[A], timeout: Option[FiniteDuration] = None): TimedTask[A] =
+    TimedTask(_ => task, timeout)
+
+}
+
+trait TaskTypes {
+  type _task[R] = |=[TimedTask, R]
+  type _Task[R] = <=[TimedTask, R]
+}
+
+trait TaskCreation extends TaskTypes {
+
+  final def taskWithContext[R: _task, A](c: ScheduledExecutorService => Task[A],
+                                           timeout: Option[FiniteDuration] = None): Eff[R, A] =
+    TimedTask(c, timeout).send
+
+  final def ofTask[R: _task, A](task: Task[A], timeout: Option[FiniteDuration] = None): Eff[R, A] =
+    TimedTask(_ => task, timeout).send[R]
+
+  final def taskFailed[R: _task, A](t: Throwable): Eff[R, A] =
+    TimedTask(_ => Task.fromTry[A](Failure(t))).send
+
+  final def taskSuspend[R: _task, A](task: => Task[Eff[R, A]], timeout: Option[FiniteDuration] = None): Eff[R, A] =
+    TimedTask(_ => Task.suspend(task), timeout).send.flatten
+
+  final def taskDelay[R: _task, A](call: => A, timeout: Option[FiniteDuration] = None): Eff[R, A] =
+    TimedTask(_ => Task.delay(call), timeout).send
+
+  final def taskForkScheduler[R: _task, A](call: Task[A], scheduler: Scheduler, timeout: Option[FiniteDuration] = None): Eff[R, A] =
+    TimedTask(_ => Task.fork(call, scheduler), timeout).send
+
+  final def taskFork[R: _task, A](call: Task[A], timeout: Option[FiniteDuration] = None): Eff[R, A] =
+    TimedTask(_ => Task.fork(call), timeout).send
+
+  final def taskAsync[R: _task, A](callbackConsumer: ((Throwable Either A) => Unit) => Unit,
```

Comment:
I like a type aliases for this like `type Callback[A] = (Throwable Either A) => Unit` and `type Subscribe[A] = Callback[A] => Unit`.